src/confidential_ml_utils/logging.py

```
@@ -54,7 +54,16 @@ class ConfidentialLogger(logging.getLoggerClass()):  # type: ignore
     def __init__(self, name: str):
         super().__init__(name)  # type: ignore
 
-    def _log(self, level, msg, category, args, **kwargs):
+    def _log(self, level, msg, args, **kwargs):
+        """
+        Note: Signature here matches superclass `_log` to avoid mismatch issues.
+        The `category` is thus assumed to be in `kwargs` and is popped
+        """
+        try:
```

Two comments here:
1) I don't think we want to raise an error. If someone calls log that is not part of our ecosystem, we should happily just not prefix the log entry without raising an error; i.e. behave the same way as DataCategory.PRIVATE
2) I believe you can give pop a default value so that you don't need the try except